IOS-3919 Update WalletCore to 3.2.1-tangem2

ba1ashov opened the pull request.

ba1ashov committed:
IOS-3919 Update WalletCore to 3.2.1-tangem1

ba1ashov committed:
IOS-3919 Adapt TON to new version

IOS-3919 Update TONTransactionBuilder.swift

ba1ashov committed:
IOS-3919 Adapt Cosmos to new version(remove external sign)

ba1ashov commented:
TON сейчас будем тестировать с @skibinalexander

megakoko commented on BlockchainSdk/WalletManagers/Cosmos/CosmosWalletManager.swift:
```diff
@@ -145,16 +149,15 @@ class CosmosWalletManager: BaseManager, WalletManager {
                     gas: nil,
-                    params: nil
+                    params: nil,
+                    signature: Data()
```
Раньше там внутри подписывалось ключем, состоящим из единиц, эта подпись внутрь включалась

Balashov152 replied:
сделал dummy signature ✅

megakoko commented on BlockchainSdk/WalletManagers/TON/TONTransactionBuilder.swift:
```diff
@@ -64,7 +64,7 @@ final class TONTransactionBuilder {
     private func input(amount: Amount, destination: String, params: TONTransactionParams?) throws -> TheOpenNetworkSigningInput {
         let transfer = try self.transfer(amount: amount, destination: destination, params: params)
-        
+
```
Лишние изменения

Balashov152 replied:
✅

megakoko commented on BlockchainSdkTests/Cosmos/CosmosTests.swift:
```diff
@@ -28,19 +28,34 @@ class CosmosTests: XCTestCase {
-        let transactionString = String(data: transactionData, encoding: .utf8)!
-        
+
+        XCTAssertEqual(dataForSign.hex, "8a6e6f74625fd39707843360120874853cc0c1d730b087f3939f4b187c75b907")
+        var signature = try XCTUnwrap(privateKey.sign(digest: dataForSign, curve: cosmosChain.coin.curve))
+        // We should delete last byte from signature
+        signature = signature.dropLast(1)
```
Хм, а зачем это в тестах? Не нашел в основном коде

Давай унесем в buildforSend, мы будем передавать подпись, а дальше функция сама сделает что надо и впендюрит ее в output

megakoko replied:
Тут можно проверку подписи убрать, она в общем-то проверится как часть output-a

megakoko replied:
Окей

megakoko commented on BlockchainSdkTests/Cosmos/CosmosTests.swift:
```diff
@@ -28,19 +28,34 @@ class CosmosTests: XCTestCase {
+
+        XCTAssertEqual(signature.hex, "f9e1f4001657a42009c4eb6859625d2e41e961fc72efd2842909c898e439fc1f549916e4ecac676ee353c7d54c5ae30a29b4210b8bff0ebfdcb375e105002f47")
+
+        let transactionData = try txBuilder.buildForSend(
+            amount: Amount(with: cosmosChain.blockchain, value: 0.000001),
+            source: wallet.address,
+            destination: "cosmos1zt50azupanqlfam5afhv3hexwyutnukeh4c573",
+            feeAmount: 0.000200,
+            gas: 200_000,
+            params: nil,
+            signature: signature
+        )
+
+        let transactionString = try XCTUnwrap(String(data: transactionData, encoding: .utf8))
```
Не очень удобное API получается, надо два раза дублировать данные. Может возвращать из buildForSign и хеш, и input? Потом скармливать в buildForSend input и подпись

megakoko replied:
Так уже и рефакторим, давай сделаем чтобы два раза не создавать input

Balashov152 replied:
я уже боюсь что то лишнее трогать, и делать лучше
потом ревертить приходится(

megakoko replied:
Ну тут же задача на рефакторинг, все норм, уместно. Как раз это место и трогаем

Balashov152 replied:
> Может возвращать из buildForSign и хеш, и input?

Не хочется так делать, так как хочется придти к однообразию, что бы для всех билдеров возвращалась Data на подпись, а то сейчас для WalletCore будет Input, а если вдруг другая либа, то там будет что то еще. И опять разъедется, надо бы вообще WalletCore в рамках/на слое TransactionBuilder держать, и все. Вдруг потом решим еще на что то перейти, и все, замучаемся выпиливать из всех файлов.

megakoko commented on BlockchainSdkTests/TON/TONTests.swift:
```diff
@@ -51,7 +51,7 @@ class TONTests: XCTestCase {
             )
             
-            XCTAssertEqual(try! input.jsonString(), "{\"privateKey\":\"hfyhNLP+P9Uj2LUoYI2AOJDibJPIbcPZe41Zx7NUDJc=\",\"transfer\":{\"walletVersion\":\"WALLET_V4_R2\",\"dest\":\"EQAoDMgtvyuYaUj-iHjrb_yZiXaAQWSm4pG2K7rWTBj9eOC2\",\"amount\":\"1000000000\",\"mode\":3,\"bounceBehavior\":\"NON_BOUNCEABLE\"}}"
+            XCTAssertEqual(try! input.jsonString(), "{\"privateKey\":\"hfyhNLP+P9Uj2LUoYI2AOJDibJPIbcPZe41Zx7NUDJc=\",\"transfer\":{\"walletVersion\":\"WALLET_V4_R2\",\"dest\":\"EQAoDMgtvyuYaUj-iHjrb_yZiXaAQWSm4pG2K7rWTBj9eOC2\",\"amount\":\"1000000000\",\"mode\":3}}"
```
А почему так? Это значение по-умолчанию?

Balashov152 replied:
@skibinalexander сможешь отписать ?

По факту когда был на bounceBehavior, он был в json, по их новой логике, его нет в signingInput.json()

Balashov152 replied:
Вообще эти тесты мы сами добавляли, а не WC брали

megakoko replied:
У них наверно не было с bouncing транзакциями

megakoko commented on Podfile:
```diff
@@ -9,8 +9,8 @@ def common_pods
   
-  pod 'TangemWalletCore', :git => 'https://github.com/tangem/wallet-core-binaries-ios.git', :tag => '3.1.9-tangem4'
-#    pod 'TangemWalletCore', :path => '../wallet-core-binaries-ios'
+  pod 'TangemWalletCore', :git => 'https://github.com/tangem/wallet-core-binaries-ios.git' , :tag => '3.2.1-tangem1'
```
Лишний пробел

megakoko replied:
<img width="90" alt="Screenshot 2023-06-28 at 15 18 10" src="https://github.com/tangem/blockchain-sdk-swift/assets/12209839/f224a9dd-4a2b-4ba9-af5a-a692c1e74bba">

Balashov152 replied:
✅

megakoko reviewed (commented):
Вообще красиво

ba1ashov committed:
IOS-3919 Refactoring CosmosTransactionBuilder to work Transaction struct

Balashov152 commented on BlockchainSdk/WalletManagers/Cosmos/CosmosTransactionBuilder.swift:
```diff
@@ -27,8 +27,53 @@ class CosmosTransactionBuilder {
+        let signatures = DataVector()
+        signatures.add(data: signature)
+
+        let compileWithSignatures = TransactionCompiler.compileWithSignatures(
```
✅

Balashov152 commented on BlockchainSdk/WalletManagers/Cosmos/CosmosTransactionBuilder.swift:
```diff
@@ -27,8 +27,53 @@ class CosmosTransactionBuilder {
+        publicKeys.add(data: wallet.publicKey.blockchainKey)
+
+        let signatures = DataVector()
+        signatures.add(data: signature)
```
Неа, только добавлять

Balashov152 commented on BlockchainSdk/WalletManagers/Cosmos/CosmosTransactionBuilder.swift:
```diff
@@ -27,8 +27,53 @@ class CosmosTransactionBuilder {
+        let output = try CosmosSigningOutput(serializedData: compileWithSignatures)
+
+        if output.error != .ok {
+            assertionFailure("\(output.errorMessage)")
```
убрал ✅

ba1ashov committed:
IOS-3919 Fix review

IOS-3919 Update Podfile

IOS-3919 Update CosmosTransactionBuilder.swift

IOS-3919 Update Podfile.lock

IOS-3919 Update CosmosTransactionBuilder.swift

ba1ashov committed:
IOS-3919 Revert "TONTests.swift"

Balashov152 commented on BlockchainSdkTests/Cosmos/CosmosTests.swift:
```diff
@@ -27,20 +27,28 @@ class CosmosTests: XCTestCase {
         txBuilder.setSequenceNumber(8)
-        
-        let input = try! txBuilder.buildForSign(
+
+        let transaction = Transaction(
```
Так галочка в Xcode стоит, вроде обсуждали что включаем ее )

megakoko replied:
Нет. Сейчас с пробелами остается все как прежде. Потом добавим обратно удаленные пробелы, отключим правило в swiftformat, и оставим стандартное поведение xcode

Balashov152 replied:
вернул ✅

ba1ashov committed:
IOS-3919 Upate TangemWalletCore to 3.2.1-tangem2

ba1ashov committed:
IOS-3919 Return whitespaces in empty lines

IOS-3919 Update CosmosTests.swift

ba1ashov commented:
> TON сейчас будем тестировать с @skibinalexander

Все ок, фиксанул extrenalSigner, все encodedTx остались прежними.

ba1ashov commented:
можно ставить галки 🐦

megakoko commented:
В tontests изменения JSON-а остались

ba1ashov commented:
> В tontests изменения JSON-а остались

Видать WalletCore сериализация в json не учитывает это поле, я хз, его там просто нет в их сериализации

megakoko commented:
Я к тому, что раньше-то было это поле. Сейчас нет. Если нода воспринимает пустоту как нужное нам значение, то окей.

ba1ashov commented:
> Я к тому, что раньше-то было это поле. Сейчас нет. Если нода воспринимает пустоту как нужное нам значение, то окей.

А, так этот json не используется для отправки куда либо, он просто для сверки input, в API отправляется только encoded hex
